src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -331,6 +332,46 @@ private CertificateInfo importCertificate(byte[] certificateBytes, boolean isFro
         return certificateInfo;
     }
 
+    /**
+     * Activates certificate by given certificateId
+     *
+     * @param certificateId
+     * @throws CertificateNotFoundException
+     */
+    public void activateCertificate(String certificateId) throws CertificateNotFoundException {
+        try {
+            signerProxyFacade.activateCert(certificateId);
+        } catch (CodedException e) {
+            if (isCausedByCertNotFound(e)) {
+                throw new CertificateNotFoundException("Certificate with id " + certificateId + " " + NOT_FOUND);
+            } else {
+                throw e;
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("certificate activation failed", e);
+        }
+    }
+
+    /**
+     * Deactivates certificate by given certificateId
+     *
+     * @param certificateId
+     * @throws CertificateNotFoundException
+     */
+    public void deactivateCertificate(String certificateId) throws CertificateNotFoundException {
+        try {
+            signerProxyFacade.deactivateCert(certificateId);
```

Comment:
Should also check for business logic rules for "is deactivate enabled".

This is defined in the old UI in https://github.com/nordic-institute/X-Road/blob/1b453f07225d5b279c4a9a60a82cdf4354fd6102/src/proxy-ui/public/javascripts/keys.js#L47-L55 and is not validated in old controllers at all.

From that, the rule for "deactivate possible" seems to be
1. currently activated
2. cert is _saved to configuration_

It probably is best to implement this in https://jira.niis.org/browse/XRDDEV-806. Service implementation will add something like this https://github.com/nordic-institute/X-Road/blob/8f3e4eadd92849b795779d17393ef66975bce53b/src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java#L617-L620

Frontend rendering also needs to take into account `possible_actions` for this certificate, to show or not show the button.

See https://confluence.niis.org/display/XRDDEV/Checking+allowed+operations for more